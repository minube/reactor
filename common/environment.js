import { DeviceInfo, Platform } from 'react-native';
import PKG from '../package.json';

const { NODE_ENV, REACT_APP_ENV = 'development' } = process.env;
const IS_SERVER = typeof window === 'undefined';
const IS_WEB = Platform.OS === 'web';
const IS_PRODUCTION = NODE_ENV === 'production' && NODE_ENV === REACT_APP_ENV;
const IS_MOBILE_WEB = IS_WEB && !IS_SERVER
  ? /iPhone|Android|Tablet|iPad|Mobile|iPod|Windows Phone|Blackberry/i.test(DeviceInfo.userAgent)
  : false;

if (IS_WEB && !IS_SERVER) window.reactor = { version: PKG.version };

export default {
  IS_NATIVE: !IS_WEB,
  IS_SERVER: IS_SERVER && IS_WEB,
  IS_WEB,
  IS_MOBILE_WEB,
  IS_PRODUCTION,
  IS_DEVELOPMENT: !IS_PRODUCTION,
  IS_TEST: NODE_ENV === 'test',

  NODE_ENV,

  PKG: {
    name: PKG.name,
    version: PKG.version,
  },

  REACT_APP_ENV,
};
