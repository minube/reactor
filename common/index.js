import STYLE from './style';
import THEME from './theme';

export {
  STYLE,
  THEME,
};
