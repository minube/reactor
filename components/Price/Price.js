import { array, number, oneOfType, string } from 'prop-types';
import React from 'react';
import { StyleSheet, View } from 'react-native';

import Text from '../Text';
import styles from './Price.style';

const LEFT_SYMBOLS = ['$'];

const Price = ({
  fixed, style, symbol, value,
}) => (
  <View style={StyleSheet.flatten([styles.container, value >= 1000 && styles.containerSmall])}>
    { LEFT_SYMBOLS.includes(symbol) &&
      <Text large style={StyleSheet.flatten([styles.symbol, style])}>{symbol}</Text> }
    <Text large bold style={style}>
      {
        parseFloat(Math.abs(value))
          .toFixed(fixed)
          // .replace(/\B(?=(\d{3})+(?!\d))/g, ',') // @TODO: Another option
          .replace(/(\d)(?=(\d{3})+\.)/g, '$1,')
      }
    </Text>
    { !LEFT_SYMBOLS.includes(symbol) &&
      <Text style={StyleSheet.flatten([styles.symbol, style])}>{symbol}</Text> }
  </View>
);

Price.propTypes = {
  fixed: number,
  style: oneOfType([array, number]),
  symbol: string,
  value: number,
};

Price.defaultProps = {
  fixed: 0,
  style: [],
  symbol: '$',
  value: 0,
};

export default Price;
