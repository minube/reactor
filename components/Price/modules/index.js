import format from './format';

export { format }; // eslint-disable-line
