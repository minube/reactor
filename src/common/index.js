import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
