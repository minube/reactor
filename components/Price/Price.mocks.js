import { THEME } from '../../common';

export default {
  FIXED: 4,
  FIXED_NULL: 0,
  LOCALE: 'es-ES',
  STYLE: {
    fontWeight: THEME.FONT.WEIGHT.BOLD,
    fontSize: THEME.UNIT * 2,
    color: THEME.COLOR.PRIMARY,
  },
  CURRENCY: 'Eur',
  OPERATOR: '+',
  SYMBOL_$: '$',
  SYMBOL_EUR: 'â‚¬',
  TEXT_PROPS: {
    headline: true,
    level: 3,
  },
  TITLE: 'From',
  VALUE: 6995.12345,
};
