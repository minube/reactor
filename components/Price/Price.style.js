import { StyleSheet } from 'react-native';

import { LAYOUT, THEME } from '../../common';

const { FONT } = THEME;

export default StyleSheet.create({
  container: LAYOUT.STYLE.ROW,

  minimize: {
    fontWeight: FONT.WEIGHT.SEMIBOLD,
    transform: [{ scale: 0.8 }],
  },
});
