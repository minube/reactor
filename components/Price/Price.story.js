import React, { Fragment } from 'react';
import { View } from 'react-native';
import { storiesOf } from '@storybook/react';
import { number, text } from '@storybook/addon-knobs/react';

import { THEME } from '../../common';
import Price from './Price';

storiesOf('✅ Price', module)
  .addWithJSX('default', () => (
    <Price />
  ))
  .addWithJSX('caption', () => (
    <Price caption="From" />
  ))
  .addWithJSX('value', () => (
    <Price value={19.95} />
  ))
  .addWithJSX('fixed', () => (
    <Price fixed={2} value={19.1234} />
  ))
  .addWithJSX('symbol (left)', () => (
    <Price value={4} symbol="$" />
  ))
  .addWithJSX('symbol (right)', () => (
    <Price value={4} symbol="eur" />
  ))
  .addWithJSX('fontSize auto-adjust', () => (
    <View>
      <Price value={32} symbol="$" />
      <Price value={1024} symbol="$" />
    </View>
  ))
  .addWithJSX('inherit <Text> properties', () => (
    <Price title primary />
  ))
  .addWithJSX('style', () => (
    <Price style={{ fontWeight: 'bold', fontSize: THEME.FONT.SIZE.SMALL, color: THEME.COLOR.ACCENT }} />
  ))
  .addWithJSX('🏀 Playground', () => (
    <Price
      caption={text('caption', 'from')}
      fixed={number('fixed', 2)}
      value={number('value', 64)}
      symbol={text('symbol', '€')}
    />
  ));

