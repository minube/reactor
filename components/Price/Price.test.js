import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Price from './Price';

const STYLE = { fontWeight: "800", fontSize: 32, color: 'green' }
const PROPS = { value: 6995.123456 };

describe('<Price>', () => {
  it('renders', () => {
    const tree = renderer.create(<Price />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {value}', () => {
    const tree = renderer.create(<Price {...PROPS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {locale} (es-ES)', () => {
    const tree = renderer.create(<Price {...PROPS} locale="es-ES" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {fixed} (4)', () => {
    const tree = renderer.create(<Price {...PROPS} fixed={4}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {fixed} (0)', () => {
    const tree = renderer.create(<Price {...PROPS} fixed={0} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {symbol} (left)', () => {
    const tree = renderer.create(<Price {...PROPS} symbol="$" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {symbol} (right)', () => {
    const tree = renderer.create(<Price {...PROPS} symbol="eur" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {title}', () => {
    const tree = renderer.create(<Price {...PROPS} title="from" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('inject <Text> properties', () => {
    const tree = renderer.create(<Price {...PROPS} headline={false} subtitle level={2}  symbol="eur" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<Price {...PROPS} style={STYLE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

