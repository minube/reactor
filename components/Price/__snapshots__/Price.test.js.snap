// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Price> inject <Text> properties 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 18,
        },
        false,
        Object {},
      ]
    }
  >
    6.995,12
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "fontWeight": "600",
          "lineHeight": 18,
        },
        false,
        Object {
          "transform": Array [
            Object {
              "scale": 0.8,
            },
          ],
        },
      ]
    }
  >
    eur
  </Text>
</View>
`;

exports[`<Price> renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {},
      ]
    }
  >
    0
  </Text>
</View>
`;

exports[`<Price> when {fixed} (0) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {},
      ]
    }
  >
    6.995
  </Text>
</View>
`;

exports[`<Price> when {fixed} (4) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {},
      ]
    }
  >
    6.995,1235
  </Text>
</View>
`;

exports[`<Price> when {locale} (es-ES) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {},
      ]
    }
  >
    6.995,12
  </Text>
</View>
`;

exports[`<Price> when {style} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "color": "green",
          "fontSize": 32,
          "fontWeight": "800",
        },
      ]
    }
  >
    6.995,12
  </Text>
</View>
`;

exports[`<Price> when {symbol} (left) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "transform": Array [
            Object {
              "scale": 0.8,
            },
          ],
        },
      ]
    }
  >
    $
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {},
      ]
    }
  >
    6.995,12
  </Text>
</View>
`;

exports[`<Price> when {symbol} (right) 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {},
      ]
    }
  >
    6.995,12
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "transform": Array [
            Object {
              "scale": 0.8,
            },
          ],
        },
      ]
    }
  >
    eur
  </Text>
</View>
`;

exports[`<Price> when {title} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {
          "transform": Array [
            Object {
              "scale": 0.8,
            },
          ],
        },
      ]
    }
  >
    from
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {},
      ]
    }
  >
    6.995,12
  </Text>
</View>
`;

exports[`<Price> when {value} 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontSize": 16,
          "lineHeight": 30,
        },
        false,
        Object {},
      ]
    }
  >
    6.995,12
  </Text>
</View>
`;
